core/client/app/components/gh-profile-image.js

```diff
@@ -28,12 +29,25 @@ export default Ember.Component.extend({
         return `background-image: url(${url})`.htmlSafe();
     }),
 
-    imageBackground: Ember.computed('email', 'size', function () {
-        var email = this.get('email'),
+    trySetValidEmail: function () {
+        var email = this.get('email');
+        if (validator.isEmail(email)) {
+            this.set('validEmail', email);
+        } else {
+            this.set('validEmail', '');
+        }
+    },
+
+    onEmailChange: Ember.observer('email', function () {
```

Comment:
Rather than using an observer here it's a bit cleaner to hook into the component lifecycle hooks, e.g.:

```
didReceiveAttrs: function () {
    Ember.run.debounce(this, 'trySetValidEmail', 700);
},
```

This has the added benefit of running immediately on first-render which observers won't always do.